### PR TITLE
Allow references for responses fixes #84

### LINKED ISSUE #84: Allow references for responses
I would like to be able to reuse responses by setting references, like this:
```
"paths": {
  "/path.json": {
    "get": {
      "resource": {
        "401": {
          "$ref": "#/responses/NotLoggedIn"
        }
      }
    }
  },
"responses": {
  "NotLoggedIn": {
    "schema": {
      "$ref": "#/definitions/Error"
    }
  }
```

Currently, it ignores the reference, letting any value pass. My current attempt is to fix this is by altering Apivore::Swagger.each_response:
```
method_data.responses.each do |response_code, response_data|
  schema_location = nil
  if response_data.schema
    schema_location = Fragment.new ['#', 'paths', path, verb, 'responses', response_code, 'schema']
  elsif response_data.send(:$ref)
    schema_location = Fragment.new response_data.send(:$ref).split('/').push('schema')
  end
  block.call(path, verb, response_code, schema_location)
end
```

But this raises "NoMethodError: undefined method `validate' for {"$ref"=>"#/definitions/Error"}:Hash"

- /.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/json-schema-2.6.1/lib/json-schema/validator.rb:114:in `validate'
- /.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/json-schema-2.6.1/lib/json-schema/validator.rb:276:in `fully_validate'
- /.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/apivore-1.6.0/lib/apivore/swagger_checker.rb:26:in `has_matching_document_for'
- /.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/apivore-1.6.0/lib/apivore/validator.rb:95:in `check_response_is_valid'
- /.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/apivore-1.6.0/lib/apivore/validator.rb:65:in `post_checks'
- /.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/apivore-1.6.0/lib/apivore/validator.rb:28:in `matches?'
- ./spec/requests/api_spec.rb:18:in 'block (3 levels) in <top (required)>'

Any plans for supporting this?

### PR DESCRIPTION
**Pull request by trekdemo**
fixes #84

### CODE REVIEW

**Comment by trekdemo** [comment]
@westfield-labs are you still maintaining this project?

**Comment by trekdemo** [comment]
@hornc Sorry to bother you, but I saw you did the latest merge. Any thoughts on this?

**Comment by ashoda** [comment]
@trekdemo thanks for submitting this PR. We will review https://github.com/westfieldlabs/apivore/issues/84 and this PR shortly.

File: lib/apivore/swagger.rb [hunk 1]
```diff
@@ -34,11 +34,16 @@ def each_response(&block)
...
-            schema_location = nil
-            if response_data.schema
-              schema_location = Fragment.new ['#', 'paths', path, verb, 'responses', response_code, 'schema']
-            end
+            schema_location =
+              case
+              when response_data.schema
+                Fragment.new(['#', 'paths', path, verb, 'responses', response_code, 'schema'])
+              when (ref = response_data['$ref'])
+                Fragment.new(ref.split('/'))
```
Comment (lucasrenan) [inline lib/apivore/swagger.rb:44]: what's the result here? seems you are missing `'schema'`

```ruby
Fragment.new(ref.split('/').push('schema'))
```

Comment (trekdemo) [reply lib/apivore/swagger.rb:44]: @lucasrenan I'm not sure why would we missing a `'schema'`. `ref` holds the value of the path to the schema definition. Eg.: in the added test it's `'#/definitions/service'`. And that's exactly the path to the schema definition at the end of file:  https://github.com/westfieldlabs/apivore/pull/89/files#diff-65c886d34e6323742cba1cd52731ecc5R129

**Comment by trekdemo** [comment]
@ashoda any update on the review?

**Pull request closed without merging**